File: C/ex05.c
#include <stdio.h>

int main(){
    int i;
    float peso_final, nota_final, nota, peso;
    peso_final = 0;
    nota_final = 0;

    for(i=0; i<=2;i++){
        printf("Digite a nota: ");
        scanf("%f", &nota);
        printf("Digite o peso: ");
        scanf("%f", &peso);

        peso_final = peso_final + peso;
        nota_final = nota_final + (peso * nota);
    }

    printf("A media ponderada foi: %.2f", nota_final/peso_final);
    return 0;
}

File: Haskell/ex05.hs
module Main where

import System.IO ()

mediaPonderada :: Float -> Float -> Float -> Float -> Float -> Float -> Float
mediaPonderada nota1 peso1 nota2 peso2 nota3 peso3 =
    (nota1 * peso1 + nota2 * peso2 + nota3 * peso3) / (peso1 + peso2 + peso3)

main :: IO ()
main = do
    putStrLn "Digite a primeira nota:"
    nota1 <- readLn
    putStrLn "Digite o peso da primeira nota:"
    peso1 <- readLn

    putStrLn "Digite a segunda nota:"
    nota2 <- readLn
    putStrLn "Digite o peso da segunda nota:"
    peso2 <- readLn

    putStrLn "Digite a terceira nota:"
    nota3 <- readLn
    putStrLn "Digite o peso da terceira nota:"
    peso3 <- readLn

    let media = mediaPonderada nota1 peso1 nota2 peso2 nota3 peso3
    putStrLn ("A média ponderada das notas é: " ++ show media)
